### PR TITLE
Move provider's log task handlers to the provider package

### PR DESCRIPTION
**Pull request by ephraimbuddy**
---
This PR fixes some of the issues listed in #9386
`CloudwatchTaskHandler`, `StackdriverTaskHandler`, and  `ElasticsearchTaskHandler` were all moved from `airflow/utils/log/` to their respective providers package

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Review by turbaszek — APPROVED** [review]
LGTM, can you please take a look at the docs error?

**Comment by ephraimbuddy** [comment]
Hi @turbaszek , please take a look.

**Review by turbaszek — APPROVED** [review]

File: UPDATING.md [hunk 1]
```diff
@@ -61,6 +61,9 @@ More tips can be found in the guide:
 https://developers.google.com/style/inclusive-documentation
 
 -->
+### CloudwatchTaskHandler have moved
+The `CloudwatchTaskHandler` class from `airflow.utils.log.cloudwatch_task_handler` has been moved to
+`airflow.providers.amazon.aws.log.cloudwatch_task_handler`. This is because it has items specific to `aws`
```
Comment (feluelle) [inline UPDATING.md]:
```suggestion
`airflow.providers.amazon.aws.log.cloudwatch_task_handler`. This is because it has items specific to `aws`.

```

**Comment by turbaszek** [comment]
@ephraimbuddy please rebase, I've just merged this fix in #9628

**Review by turbaszek — APPROVED** [review]

File: airflow/utils/log/cloudwatch_task_handler.py [hunk 2]
```diff
@@ -15,101 +15,15 @@
...
+"""
+import warnings
+
+# pylint: disable=unused-import
+from airflow.providers.amazon.aws.log.cloudwatch_task_handler import CloudwatchTaskHandler  # noqa
+
+warnings.warn(
+    "This module is deprecated. Please use `airflow.providers.amazon.aws.log.cloudwatch_task_handler`.",
+    DeprecationWarning, stacklevel=2
+)
```
Comment (turbaszek) [inline airflow/utils/log/cloudwatch_task_handler.py:29]: Would you mind adding a test in `test_core_to_contrib.py`?

**Comment by ephraimbuddy** [comment]
Hi @turbaszek, I want to combine #9624 and #9623 into this PR so it can be merged at once. WDYT?

**Comment by mik-laj** [comment]
@ephraimbuddy  +1 That shouldn't be a problem for us.

File: airflow/providers/google/cloud/log/stackdriver_task_handler.py [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+        self.task_instance_labels = self._task_instance_to_labels(task_instance)
+
+    def read(
+        self, task_instance: TaskInstance, try_number: Optional[int] = None, metadata: Optional[Dict] = None
+    ) -> Tuple[List[str], List[Dict]]:
+        """
+        Read logs of given task instance from Stackdriver logging.
+
+        :param task_instance: task instance object
+        :type: task_instance: TaskInstance
```
Comment (ephraimbuddy) [inline airflow/providers/google/cloud/log/stackdriver_task_handler.py]:
```suggestion
        :type: task_instance: :class: airflow.models.TaskInstance
```

File: airflow/providers/google/cloud/log/stackdriver_task_handler.py [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+        else:
+            labels = None
+        self._transport.send(record, message, resource=self.resource, labels=labels)
+
+    def set_context(self, task_instance: TaskInstance) -> None:
+        """
+        Configures the logger to add information with information about the current task
+
+        :param task_instance: Currently executed task
+        :type task_instance: TaskInstance
```
Comment (ephraimbuddy) [inline airflow/providers/google/cloud/log/stackdriver_task_handler.py]:
```suggestion
        :type task_instance:  :class: airflow.models.TaskInstance
```

File: airflow/providers/google/cloud/log/stackdriver_task_handler.py [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+        self.task_instance_labels = self._task_instance_to_labels(task_instance)
+
+    def read(
+        self, task_instance: TaskInstance, try_number: Optional[int] = None, metadata: Optional[Dict] = None
+    ) -> Tuple[List[str], List[Dict]]:
+        """
+        Read logs of given task instance from Stackdriver logging.
+
+        :param task_instance: task instance object
+        :type: task_instance: :class: airflow.models.TaskInstance
```
Comment (ephraimbuddy) [inline airflow/providers/google/cloud/log/stackdriver_task_handler.py]:
```suggestion
        :type task_instance: :class: airflow.models.TaskInstance
```

**Review by turbaszek — APPROVED** [review]

**Pull request merged**